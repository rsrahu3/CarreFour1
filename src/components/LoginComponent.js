import * as React from 'react';
import history from './common/history';
import TopHeader from  './common/TopHeader';

export default class LoginComponent extends React.Component {
  formSubmit = (e)=>{
    history.push("/home");
  }
  render () {
    return (<div id="login">
    <TopHeader/>
    <div className="form">
      <form onSubmit={this.formSubmit} style={{ width: '30%', padding: '20px' }}>
        <div className="form-group">
          <label>Email address:</label>
          <input type="email" required className="form-control" id="email"/>
        </div>
        <div className="form-group">
          <label>Password:</label>
          <input type="password" required className="form-control" id="password"/>
        </div>
        <button type="submit" className="btn btn-default">Submit</button>
      </form>
    </div>
    </div>
    )
  }
}
