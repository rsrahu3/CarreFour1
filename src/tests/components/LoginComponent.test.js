import React from 'react';
import {shallow,configure} from 'enzyme';
import LoginComponent from './../../components/LoginComponent';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe("Login Screen test",()=>{
    it('form check',()=>{
        const enzymeWrapper = shallow(<LoginComponent/>);
        expect(enzymeWrapper.find('form').length).toBe(1);
        expect(enzymeWrapper.find('#email').length).toBe(1);
        expect(toJson(enzymeWrapper)).toMatchSnapshot();
    })
    
})