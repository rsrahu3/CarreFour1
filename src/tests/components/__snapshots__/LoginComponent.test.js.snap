// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen test form check 1`] = `
<div
  id="login"
>
  <topHeader />
  <div
    className="form"
  >
    <form
      onSubmit={[Function]}
      style={
        Object {
          "padding": "20px",
          "width": "30%",
        }
      }
    >
      <div
        className="form-group"
      >
        <label>
          Email address:
        </label>
        <input
          className="form-control"
          id="email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Password:
        </label>
        <input
          className="form-control"
          id="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn btn-default"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
